### PR TITLE
fix: Team links and username starting with team rewrites

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes # (issue)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Test A
- [ ] Test B

## Mandatory Tasks
- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/next.config.js [hunk 1]
```diff
@@ -188,7 +188,8 @@ const nextConfig = {
     // It would also not match /free/30min/embed because we are ensuring just two slashes
     // ?!book ensures it doesn't match /free/book page which doesn't have a corresponding new-booker page.
     // [^/]+ makes the RegExp match the full path, it seems like a partial match doesn't work.
-    const userTypeRouteRegExp = `/:user((?!${pages.join("|")})[^/]*)/:type((?!book)[^/]+)`;
+    const userTypeRouteRegExp = `/:user((?!${pages.join("/|")}$)[^/]*)/:type((?!book)[^/]+)`;
```
Comment (hariombalhara) [inline apps/web/next.config.js:191]: / ensures that only `team/` is excluded and not `teampro`

File: apps/web/next.config.js [hunk 2]
```diff
@@ -188,7 +188,8 @@ const nextConfig = {
...
+    const teamTypeRouteRegExp = "/team/:slug/:type((?!book)[^/]+)";
```
Comment (hariombalhara) [inline apps/web/next.config.js:192]: Ensures that for team event as well /book endpoint isn't included in rewrite.

**Review by emrysal — DISMISSED** [review]
lgtm

**Review by JeroenReumkens — DISMISSED** [review]
Tested it locally too. Works great!!

**Review by emrysal — APPROVED** [review]
lgtm 👍

**Pull request merged**